packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -0,0 +1,237 @@
+    return value;
+  }
+
+  // clean the input for any non-digit values.
+  let phoneNumber = value.replace(/[^\d*#+]/g, '');
+
+  // if phone number starts with 1, format like 1 (xxx)xxx-xxxx.
+  // if phone number starts with +, we format like +x (xxx)xxx-xxxx.
+  // For now we are only supporting NA phone number formatting with country code +x
+  // first we chop off the countrycode then we add it on when returning
+  let countryCodeNA = '';
+  if (phoneNumber[0] === '1') {
+    countryCodeNA = '1';
+
+    phoneNumber = phoneNumber.slice(1, phoneNumber.length);
+  } else if (phoneNumber[0] === '+') {
+    countryCodeNA = phoneNumber.slice(0, 2);
+    phoneNumber = phoneNumber.slice(2, phoneNumber.length);
+  }
+
+  // phoneNumberLength is used to know when to apply our formatting for the phone number
+  const phoneNumberLength = phoneNumber.length;
+
+  // we need to return the value with no formatting if its less then four digits
+  // this is to avoid weird behavior that occurs if you  format the area code too early
+  // if phoneNumberLength is greater than 10 we don't do any formatting
+
+  if (phoneNumberLength < 4 || phoneNumberLength > 10) {
+    return countryCodeNA + phoneNumber;
+  }
+
+  // if phoneNumberLength is greater than 4 and less the 7 we start to return
+  // the formatted number
+  if (phoneNumberLength < 7) {
+    return `${countryCodeNA}(${phoneNumber.slice(0, 3)})${phoneNumber.slice(3)}`;
+  }
+
+  // finally, if the phoneNumberLength is greater then seven, we add the last
+  // bit of formatting and return it.
+  return `${countryCodeNA}(${phoneNumber.slice(0, 3)})${phoneNumber.slice(3, 6)}-${phoneNumber.slice(
+    6,
+    phoneNumber.length
+  )}`;
+};
+
+const DialpadContainer = (props: {
+  errorText: string;
+  defaultText: string;
+  containerStyles?: IStyle;
+  rowStyles?: IStyle;
+  textFieldStyles?: Partial<ITextFieldStyles>;
+  digitStyles?: IStyle;
+  subStyles?: IStyle;
+}): JSX.Element => {
+  const theme = useTheme();
+  const [textValue, setTextValue] = useState('');
+  const [error, setError] = useState('');
+  const onClickDialpad = (input: string): void => {
+    setError('');
+
+    setTextValue(textValue + input);
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const setText = (e: any): void => {
+    // if text content includes non-valid dialpad input return an error
+    if (!`${e.target.value}`.match(/^[0-9*#+()-]*$/)) {
+      setError(props.errorText);
+    } else {
+      setError('');
+      setTextValue(e.target.value);
+    }
+  };
+
+  return (
+    <div className={mergeStyles(containerStyles(theme), props.containerStyles)}>
+      <TextField
+        styles={concatStyleSets(textFieldStyles(theme), props.textFieldStyles)}
+        value={formatPhoneNumber(textValue)}
+        onChange={setText}
+        errorMessage={error}
+        placeholder={props.defaultText}
+      />
+      <div className={mergeStyles(rowStyles(theme), props.rowStyles)}>
```

Comment:
Consider a FocusZone around the numbers of the dialpad for A11y purposes (so a user tabs to the dialpad then uses arrow keys to go up/down/left/right to keys). I'm not 100% certain this is the correct thing as a11y guidelines are sparse but this feels like a correct use, perhaps @dmceachernmsft can confirm this behavior or not.
https://developer.microsoft.com/en-us/fluentui#/controls/web/focuszone

Reply:
updated